### PR TITLE
Added test cases for working with KnowledgeBase in Python

### PR DESCRIPTION
**Pull request by sasjonge**
This PR adds 4 tests cases to test common use cases for using KnowledgeBase in Python.

### CODE REVIEW

File: include/knowrob/terms/Atomic.h [hunk 1]
```diff
@@ -57,6 +58,17 @@ namespace knowrob {
 		 */
 		virtual std::string_view stringForm() const = 0;
 
+		/**
+		 * Get the lexical form of this atomic term.
+		 * @return the lexical form of this atomic term.
+		 */
+		std::string humanReadableForm() const {
```
Comment (daniel86) [inline include/knowrob/terms/Atomic.h:65]: Any reason you did not add this to `Term` instead? IMO it should be a method of Term

File: src/integration/InterfaceUtils.cpp [hunk 2]
```diff
@@ -19,6 +19,19 @@
 
 using namespace knowrob;
 
+// enum for epistemicOperator
+enum EpistemicOperator {
```
Comment (daniel86) [inline src/integration/InterfaceUtils.cpp:23]: It might be confusing that we also have the enum `ModalType` in `ModalOperator.h`.
Can't we unify these?

File: src/integration/InterfaceUtils.cpp [hunk 3]
```diff
@@ -80,7 +93,7 @@ InterfaceUtils::applyModality(const std::unordered_map<std::string, boost::any>
 
 	// Retrieve epistemicOperator and check if it is "BELIEF"
 	auto epistemicOperator = boost::any_cast<int>(options.at("epistemicOperator"));
-	if (epistemicOperator == 1) {
+	if (epistemicOperator == EpistemicOperator::BELIEF) {
```
Comment (daniel86) [inline src/integration/InterfaceUtils.cpp:96]: I am a bit confused about the whole "applyModalOperator" thing. Maybe you can explain to me offline?

File: src/integration/python/knowrob_module.cpp [hunk 4]
```diff
@@ -156,5 +194,6 @@ BOOST_PYTHON_MODULE (MODULENAME) {
 	// mappings for static functions
 	def("InitKnowledgeBaseWithArgs", &InitKnowledgeBaseWrapper, "Initialize the Knowledge Base with arguments.");
 	def("InitKnowledgeBase", &InitKnowledgeBaseFromSysArgv, "Initialize the Knowledge Base using sys.argv.");
+	def("applyModality", &applyModalityWrapper, "Apply a modality to a formula.");
```
Comment (daniel86) [inline src/integration/python/knowrob_module.cpp]: Why is this here and not in InterfaceUtils.cpp?

File: src/terms/Atomic.cpp [hunk 5]
```diff
@@ -85,6 +85,9 @@ namespace knowrob::py {
 		class_<Atomic, std::shared_ptr<AtomicWrap>, bases<Term>, boost::noncopyable>
 				("Atomic", no_init)
 				.def("stringForm", pure_virtual(&Atomic::stringForm))
+				.def("__repr__", &Atomic::stringForm)
```
Comment (daniel86) [inline src/terms/Atomic.cpp:90]: better move into Term class? also below

File: src/terms/Term.cpp [hunk 6]
```diff
@@ -109,5 +109,7 @@ namespace knowrob::py {
 				.def("isBlank", &Term::isBlank)
 				.def("isGround", &Term::isGround)
 				.def("variables", pure_virtual(&Term::variables), return_value_policy<copy_const_reference>());
+		// register shared_ptr to Term
+		register_ptr_to_python< std::shared_ptr< Term > >();
```
Comment (daniel86) [inline src/terms/Term.cpp:114]: I think this is not needed. the pointer is registered in the `class_` statement or not?

File: tests/py/test_boost_python.py [hunk 7]
```diff
@@ -5,6 +5,30 @@
...
+		"temporalOperator": TemporalOperator.CURRENTLY,
+		"minPastTimestamp": -1.0,
+		"maxPastTimestamp": -1.0,
+	}
+	# Load the settings
+	kb = KnowledgeBase(settings_path)
+	# Create a formula for the query
+	phi = QueryParser.parse(query_string)
+	# Apply the modality
+	mPhi = applyModality(modalities, phi)
```
Comment (daniel86) [inline tests/py/test_boost_python.py]: Why don't you put the modal operator into the query string instead? Then you do not need to apply a modal operator after parsing.

Also there is already a syntax available for doing this. You can find an example here: https://github.com/knowrob/knowrob/blob/dev/tests/KnowledgeBaseTest.cpp#L273

**Review by daniel86 — CHANGES_REQUESTED** [review]

File: src/integration/InterfaceUtils.cpp (hunk 2 above)
Comment (sasjonge) [reply src/integration/InterfaceUtils.cpp:23]: As discussed: It would be nicer for the ROS messages (and also future REST definitions etc.) to have each modaltype start counting with 0

File: src/integration/InterfaceUtils.cpp (hunk 3 above)
Comment (sasjonge) [reply src/integration/InterfaceUtils.cpp:96]: As discussed: This is necessary for ROS or other interfaces that provide modalities in external map like structures

File: src/terms/Term.cpp (hunk 6 above)
Comment (sasjonge) [reply src/terms/Term.cpp:114]: I tested it again without this line, but then the kb_positive_query test fails in line 136 when getting the substitution.

File: src/knowrob.cpp [hunk 8]
```diff
@@ -78,11 +80,61 @@ namespace knowrob {
...
+
+namespace knowrob::py {
+	void staticKnowRobModuleInit() {
+		using namespace boost::python;
+		using namespace knowrob;
+
+		/////////////////////////////////////////////////////
+		// mappings for static functions
+		def("InitKnowledgeBaseWithArgs", &InitKnowledgeBaseWrapper, "Initialize the Knowledge Base with arguments.");
+		def("InitKnowRob", &InitKnowledgeBaseFromSysArgv, "Initialize the Knowledge Base using sys.argv.");
```
Comment (daniel86) [inline src/knowrob.cpp]: here is a minor name missmatch

**Review by daniel86 — APPROVED** [review]
